### DISCUSSION TITLE
Different behavior from MMFF minimization with added torsion constraint

### DISCUSSION THREAD

**Discussion by jasondbiggs** [rdkit/rdkit#3780] [Q&A] [answered]
In this code snippet I just use an added torsion constraint with a very high force constant to get certain dihedral angle.  It works well in version "2019.03.3" but gives very bad results in "2020.09.1".

Is this a bug or is my code flawed?

```
m = Chem.rdmolfiles.MolFromMolFile('/Users/jasonb/butane.mol',removeHs = False)
mp = AllChem.MMFFGetMoleculeProperties(m)
ff = AllChem.MMFFGetMoleculeForceField(m, mp)
ff.MMFFAddTorsionConstraint(0,1,2,3, False, -180, -180, 10000.0)
ff.Minimize()
Chem.rdmolfiles.MolToMolFile(m,'/Users/jasonb/butane_180.mol')
```
Starting with the attached MOL file, rdkit version "2019.03.3" gives the top result, while version "2020.09.1" gives the bottom version:

<img width="286" alt="image" src="https://user-images.githubusercontent.com/16337306/106640656-ec84a280-654b-11eb-9994-c477e9f11d67.png">

(can't seem to attach a .mol file, renamed to 'butane.txt')

[butane.txt](https://github.com/rdkit/rdkit/files/5913233/butane.txt)

edit:

To add another example, using the same MOL file and this code
```
m = Chem.rdmolfiles.MolFromMolFile('/Users/jasonb/butane.txt',removeHs = False)
mp = AllChem.MMFFGetMoleculeProperties(m)
globalFF = AllChem.MMFFGetMoleculeForceField(m, mp)
m2 = m;
res = []
for x in range(-180,180,5):
	ff2 = AllChem.MMFFGetMoleculeForceField(m2, mp)
	ff2.MMFFAddTorsionConstraint(0,1,2,3, False, x - .1, x + .1, 100000.0)
	ff2.Minimize()
	res.append(globalFF.CalcEnergy(ff2.Positions()))
res
```
I was previously able to get a nice plot like this,
<img width="307" alt="image" src="https://user-images.githubusercontent.com/16337306/106652423-0ed1ec80-655b-11eb-979c-260176ac21ad.png">
but it now returns a **very different** list

<img width="379" alt="image" src="https://user-images.githubusercontent.com/16337306/106652585-480a5c80-655b-11eb-8024-04a1eae90ade.png">

**Comment by jasondbiggs** [comment]
@ptosco I noticed there was this pull request, https://github.com/rdkit/rdkit/pull/2732.  Is the issue I am seeing here somehow related to that PR?

**Comment by ptosco** [comment, marked as answer]
@jasondbiggs Thanks for reporting this. I have definitely screwed it up. I have just filed this as a bug (#3781) and assigned it to myself, which is why I have closed the discussion here. I'll fix this, put proper tests in place and let you know once it's done. Sorry for the inconvenience.
